benches/serde.rs

```diff
@@ -160,30 +160,45 @@ fn make_records(record: &Value, count: usize) -> Vec<Value> {
     records
 }
 
-fn write(schema: &Schema, records: Vec<Value>) -> Vec<u8> {
+fn write(schema: &Schema, records: &[Value]) -> Vec<u8> {
     let mut writer = Writer::new(&schema, Vec::new());
-    writer.extend(records.into_iter()).unwrap();
+    writer.extend_from_slice(records).unwrap();
     writer.into_inner()
 }
 
-fn read(schema: &Schema, bytes: Vec<u8>) {
-    let reader = Reader::with_schema(schema, &bytes[..]);
+fn read<'a, I: Into<Option<&'a Schema>>>(schema: I, bytes: &[u8]) {
+    let schema = schema.into();
```

Comment:
Can I ask you to check how much this conversion and the if below impact on the benchmark? I _think_ they should be very cheap, but if they turned out to be not negligible then we might want to separate the two functions, even if it turns to be much uglier.

Reply:
I'll experiment with it. I would suspect it should be a pretty small cost, given that `Option<&T>` satisfies the null pointer optimization. I'll report back soon.

Reply:
It doesn't seem to make a difference on the benchmarks, but I think it looks cleaner to have two functions even though they are largely the same.